app/controllers/revisions_controller.rb

```diff
@@ -40,6 +40,44 @@ def show
             )
   end
 
+  def trash
+    # Get all versions whose event is destroy.
+    @revisions = PaperTrail::Version.where(event: 'destroy').order(created_at: :desc)
+  end
+
+  def recover
+    revision = PaperTrail::Version.find params[:id]
+    object = revision.reify
+    # If object's node was destroyed, assign it no a new node.
+    if !Node.exists?(object.node_id)
+      recovered_node = Node.create(label: 'Recovered', type_id: Node::Types::DEFAULT)
+      object.node_id = recovered_node.id
+    end
+
+    # If object is evidence and its issue doesn't exist any more, recover it.
+    if revision.item_type == 'Evidence' and !Note.exists?(object.issue_id)
+      issue_revision = PaperTrail::Version.where(event: 'destroy', item_type: 'Note', item_id: object.issue_id).limit(1).first
+      # A destroy revision should always be present, but just in case.
+      if issue_revision
+        issue_object = issue_revision.reify
+        issue_object.node_id = Node.issue_library.id
+        issue_object.save
+        object.issue_id = issue_object.id
+        issue_revision.destroy
+      end
+    end
+
+    if object.save
+      # Destroy revision so item is not listed in trash any more.
+      revision.destroy
```

Comment:
I just found [this article](https://leonid.shevtsov.me/post/how-to-use-papertrail-for-soft-deletetion/) which has a (very complicated) ActiveRecord query that we can supposedly use to get all deleted, non-recovered revisions to list here *without* having to destroy the "delete" revisions.

What do you think about this (assuming that the code there works; I haven't tested it)? I think that this is a better idea in principle for the sake of completeness - it feels icky deleting parts of the revision history.